modules/calcite/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -184,6 +220,23 @@ SqlDrop SqlDropTable(Span s, boolean replace) :
     }
 }
 
+SqlDrop SqlDropIndex(Span s, boolean replace) :
+{
+    final boolean ifExists;
+    final SqlIdentifier idxId;
+    final SqlIdentifier tblId;
+}
+{
+    <INDEX> 
+    ifExists = IfExistsOpt()
+    idxId = SimpleIdentifier()
+    <ON>
```

Comment:
ON clause is not the part of DROP command

Reply:
for now we need to know table name, check 
tableManager.alterTable(fullName
...  idxes.delete(cmd.indexName()); will discuss it